src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -2274,3 +2275,43 @@ def test_fw_comments(fortran_writer):
         "  end subroutine my_routine  ! My routine inline comment\n\n"
         "end module my_container  ! My container inline comment\n")
     assert expected == fortran_writer(container)
+
+
+def test_fw_directive_with_clause(fortran_reader, fortran_writer):
+    '''Test that a PSyIR directive with clauses is translated to
+    the required Fortran code.
+
+    '''
+    # Generate fparser2 parse tree from Fortran code.
+    code = (
+        "program test\n"
+        "  integer, parameter :: n=20\n"
+        "  integer :: i\n"
+        "  real :: a(n)\n"
+        "  do i=1,n\n"
+        "    a(i) = 0.0\n"
+        "  end do\n"
+        "end program test")
+    container = fortran_reader.psyir_from_source(code)
+    schedule = container.children[0]
+    loops = schedule.walk(Loop)
+    loop = loops[0].detach()
+    directive = OMPTaskloopDirective(children=[loop], num_tasks=32)
+    master = OMPMasterDirective(children=[directive])
+    parallel = OMPParallelDirective(children=[master])
+    schedule.addchild(parallel, 0)
+    assert '''!$omp parallel default(shared), private(i)
+  !$omp master
+  !$omp taskloop num_tasks(32)
```

Comment:
Please add nogroup clause here as well so we check that the clauses join is done with whitespaces.

Reply:
Done.